met-api/src/met_api/services/survey_service.py

```diff
@@ -14,48 +16,31 @@ class SurveyService:
     @classmethod
     def get(cls, survey_id):
         """Get survey by the id."""
-        db_data = Survey.get_survey(survey_id)
-        db_data['engagement'] = cls.supply_banner_url(db_data.get('engagement', None))
-        return db_data
+        survey_model: SurveyModel = SurveyModel.get_survey(survey_id)
+        engagement_model: EngagementModel = EngagementModel.get_engagement(survey_model.engagement_id)
+        survey = SurveySchema().dump(survey_model)
+        eng = EngagementSchema().dump(engagement_model)
+        eng['banner_url'] = ObjectStorageService.get_url(engagement_model.banner_filename)
+        survey['engagement'] = eng
+        return survey
 
     @classmethod
     def get_open(cls, survey_id):
         """Get survey by the id."""
-        db_data = Survey.get_open(survey_id)
-        db_data['engagement'] = cls.supply_banner_url(db_data.get('engagement', None))
-        return db_data
-
-    @staticmethod
-    def supply_banner_url(engagement):
-        """Supply engagement with banner url."""
-        if not engagement:
-            return None
-        engagement['banner_url'] = ObjectStorageService.get_url(engagement.get('banner_filename', None))
-        return engagement
-
-    @classmethod
-    def get_surveys(cls, unlinked=False):
-        """Get surveys."""
-        if unlinked:
-            return cls.get_unlinked()
-        return cls.get_all()
-
-    @classmethod
-    def get_all(cls):
-        """Get all surveys."""
-        db_data = Survey.get_all_surveys()
-        return db_data
-
-    @classmethod
-    def get_unlinked(cls):
-        """Get all surveys."""
-        db_data = Survey.get_all_unlinked_surveys()
-        return db_data
+        survey_model = SurveyModel.get_open(survey_id)
+        engagement_model: EngagementModel = EngagementModel.get_engagement(survey_model.engagement_id)
+        survey = SurveySchema().dump(survey_model)
+        eng = EngagementSchema().dump(engagement_model)
+        eng['banner_url'] = ObjectStorageService.get_url(engagement_model.banner_filename)
```

Comment:
could we maybe move the banner url to a method inside the schema?

Reply:
I am writing a ticket for improving this.